modules/persistence/src/services/metadata/associated_products/index.ts

```diff
@@ -109,26 +121,27 @@ const getAssociatedProducts = async (umbrellaMetadata) => {
     const snooty = await db();
 
     // This query matches on projects in associated_products for our given metadata that have a build_id and that do not have merged tocs
+    // then sorts by build_id for all matches
     // then groups per branch and per project from those matches
-    // and gets the most recent doc entry (by build_id), with the toctree and toctreeOrder fields.
-    const tocs = snooty.collection('metadata').aggregate([
+    // and gets the first metadata entry for each group
+    // since we are sorted by build_id, we get the most recent entry.
+    const associatedMetadataEntries = snooty.collection('metadata').aggregate([
       {
         $match: { project: { $in: associatedProductNames }, build_id: { $exists: true }, is_merged_toc: { $ne: true } },
```

Comment:
`build_id: $exists: true`

qq: meaning that both umbrella + associated products have to be persisted at least once to have the merged ToC available

Reply:
Yep - otherwise the module doesn't have any data to source.